### PR TITLE
Topic defect account products

### PR DESCRIPTION
**Pull request by CHenrie**
@jbest84  Needs Review
multiple defects:
13091765 [Fixed]
13091777 [Fixed]
13091667 [Fixed]
13091781 [Fixed]
13091782 [Closed]
13091788 [Fixed]
13091789 [Fixed]
13091805 [Fixed]
13091799 [Fixed]
13091808 [Fixed]

### CODE REVIEW

**Comment by slxbuilder** [comment]
Merged build triggered.

**Comment by slxbuilder** [comment]
Merged build started.

**Comment by slxbuilder** [comment]
Merged build finished.

File: src/Views/Opportunity/Detail.js [hunk 1]
```diff
@@ -181,14 +182,12 @@ define('Mobile/SalesLogix/Views/Opportunity/Detail', [
...
-                            baseRate = App.context['exchangeRates'][baseCode];
-                            convertedValue = val * baseRate;
-                            return format.multiCurrency.call(null, convertedValue, baseCode);
+                       if (App.hasMultiCurrency()) {
+                            var exhangeRate, convertedValue;
+                            exhangeRate = App.getBaseExchangeRate();
+                            convertedValue = val * exhangeRate.rate;
+                            return format.multiCurrency.call(null, convertedValue, exhangeRate.code);
                         }
-
```
Comment (jbest84) [inline src/Views/Opportunity/Detail.js:191]: Prefer we leave a blank line after blocks (if, for, etc). Perhaps we can discuss some style guidelines?

File: src/Views/Opportunity/Detail.js [hunk 2]
```diff
@@ -181,14 +182,12 @@ define('Mobile/SalesLogix/Views/Opportunity/Detail', [
...
                     property: 'SalesPotential',
                     renderer: (function(val) {
-                        var baseCode, baseRate, convertedValue;
-                        if (App.hasMultiCurrency()) {
-                            baseCode = App.context['systemOptions']['BaseCurrency'];
-                            baseRate = App.context['exchangeRates'][baseCode];
-                            convertedValue = val * baseRate;
-                            return format.multiCurrency.call(null, convertedValue, baseCode);
+                       if (App.hasMultiCurrency()) {
+                            var exhangeRate, convertedValue;
```
Comment (jbest84) [inline src/Views/Opportunity/Detail.js]: Move var back to the top of the function declaration.

File: src/Views/Opportunity/Detail.js [hunk 3]
```diff
@@ -280,14 +279,13 @@ define('Mobile/SalesLogix/Views/Opportunity/Detail', [
...
-                    property: 'SalesPotentialMine',
+                    property: 'SalesPotential',
                     renderer: (function(val) {
-                        var myCode, myRate, convertedValue;
                         if (App.hasMultiCurrency()) {
-                            myCode = App.context['userOptions']['General:Currency'];
-                            myRate = App.context['exchangeRates'][myCode];
-                            convertedValue = val.SalesPotential * myRate;
-                            return format.multiCurrency.call(null, convertedValue, myCode);
+                            var exhangeRate, convertedValue;
```
Comment (jbest84) [inline src/Views/Opportunity/Detail.js]: Move var to top of function.

File: src/Views/Opportunity/Edit.js [hunk 4]
```diff
@@ -309,20 +315,20 @@ define('Mobile/SalesLogix/Views/Opportunity/Edit', [
                     }
                 ]
             };
-
+
```
Comment (jbest84) [inline src/Views/Opportunity/Edit.js]: Unnecessary whitspace.

File: src/Views/OpportunityProduct/Detail.js [hunk 5]
```diff
@@ -158,69 +174,91 @@ define('Mobile/SalesLogix/Views/OpportunityProduct/Detail', [
                         name: 'Quantity',
                         property: 'Quantity'
                     },
-                    {
+                   
+                ]
+            };
+
+            if (!App.hasMultiCurrency()) {
+
```
Comment (jbest84) [inline src/Views/OpportunityProduct/Detail.js:181]: Remove this blank line.

File: src/Views/OpportunityProduct/Edit.js [hunk 6]
```diff
@@ -268,7 +247,7 @@ define('Mobile/SalesLogix/Views/OpportunityProduct/Edit', [
                     type: 'hidden'
                 },
                 {
-                    label: this.productText,
+                    label: this.productText,
```
Comment (jbest84) [inline src/Views/OpportunityProduct/Edit.js]: Trailing whitespace after the comma.

File: src/Views/OpportunityProduct/Edit.js [hunk 7]
```diff
@@ -291,16 +270,16 @@ define('Mobile/SalesLogix/Views/OpportunityProduct/Edit', [
                     type: 'lookup',
                     view: 'productprogram_related',
                     validator: validator.exists,
-                    where: (function(){
+                    where: (function () {
```
Comment (jbest84) [inline src/Views/OpportunityProduct/Edit.js]: We are not putting a space after the function declaration and the parens.

File: src/Views/Opportunity/Detail.js (hunk 2 above)
Comment (CHenrie) [inline src/Views/Opportunity/Detail.js]: Why would you declare them if you will not use them?

Comment (jbest84) [inline src/Views/Opportunity/Detail.js]: View the full changes. You are using them on the next line, Github just left it out from my comment.

Comment (CHenrie) [inline src/Views/Opportunity/Detail.js]: right they are only being used with in the if statement?

**Comment by jbest84** [comment]
@CHenrie Accept.

**Comment by slxbuilder** [comment]
Merged build triggered.

**Comment by slxbuilder** [comment]
Merged build started.

**Comment by slxbuilder** [comment]
Merged build finished.

**Comment by slxbuilder** [comment]
Merged build triggered.

**Comment by slxbuilder** [comment]
Merged build started.

**Comment by slxbuilder** [comment]
Merged build finished.

**Comment by slxbuilder** [comment]
Merged build triggered.

**Comment by slxbuilder** [comment]
Merged build started.

**Comment by slxbuilder** [comment]
Merged build finished.

**Comment by CHenrie** [comment]
@jbest84 Please Re Review

**Comment by jbest84** [comment]
@CHenrie Just one whitespace issue, fix that and it looks good.

**Comment by slxbuilder** [comment]
Merged build triggered.

**Comment by slxbuilder** [comment]
Merged build started.

**Comment by CHenrie** [comment]
@jbest84  Please Re Review

**Comment by jbest84** [comment]
@CHenrie Accept

**Comment by slxbuilder** [comment]
Merged build finished.

**Comment by jbest84** [comment]
Validated by QA. Merging.

**Pull request merged**